funding/commitment_type_negotiation.go

```diff
@@ -29,12 +29,32 @@ func negotiateCommitmentType(channelType *lnwire.ChannelType,
 	local, remote *lnwire.FeatureVector) (lnwallet.CommitmentType, error) {
 
 	if channelType != nil {
-		if !hasFeatures(local, remote, lnwire.ExplicitChannelTypeOptional) {
-			return 0, errUnsupportedExplicitNegotiation
+		// If the peer does know explicit negotiation, let's attempt
+		// that now.
+		if hasFeatures(local, remote, lnwire.ExplicitChannelTypeOptional) {
+			return explicitNegotiateCommitmentType(
+				*channelType, local, remote,
+			)
 		}
-		return explicitNegotiateCommitmentType(
-			*channelType, local, remote,
+
+		// If they don't know explicit negotiation, let's fall back to
+		// implicit negotiation if they just signal one of the known
+		// default types.
+		channelFeatures := lnwire.RawFeatureVector(*channelType)
+		staticRemoteKeyOnly := channelFeatures.OnlyContains(
+			lnwire.StaticRemoteKeyRequired,
+		)
+		anchorOnly := channelFeatures.OnlyContains(
+			lnwire.AnchorsZeroFeeHtlcTxRequired,
 		)
+
+		// It's one of the default types.
```

Comment:
The spec says "use the smallest bitmap possible", but leaves that somewhat open. For example, for anchors we'll set _both_ the bits above. 

This check here should be removed, causing us to fall back to implicit negotiation here has normal. Eventually, we'll require this bit and only accept an explicit channel type.